File: VisualStudio/CSharp/w7cpframe/clsPfMaterial.cs
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Text.RegularExpressions;

namespace w7cpframe
{
    class clsPfMaterial
    {



        public int key = 0;

        public double density = 0;        //.. density ..
        public double emod = 0;           //.. elastic Modulus ..
        public double therm = 0;          //.. coeff of thermal expansion..



        public void initialise()
        {
            key = 0;
            density = 0;
            emod = 0;
            therm = 0;
        }

        public void setValues(int materialKey, double massDensity, double ElasticModulus, double CoeffThermExpansion)
        {
            key = materialKey;
            density = massDensity;
            emod = ElasticModulus;
            therm = CoeffThermExpansion;
        }

        public string sprint()
        {
            string s; 

            s = "";
            s = s + key.ToString().PadLeft(8);
            s = s + density.ToString().PadLeft( 15);
            s = s + emod.ToString().PadLeft( 15);
            s = s + therm.ToString().PadLeft( 15);

            return s;

        }

        public void cprint()
        {
            System.Console.WriteLine(sprint());
        }

        public void fprint(StreamWriter fp)
        {
            fp.WriteLine(sprint());
        }

        public void fgetData(StreamReader fp)
        {
            string s;

            System.Console.WriteLine("fgetData ...");

            s = fp.ReadLine();
            //WScript.Echo( s);
            sgetData(s);


            System.Console.WriteLine("... fgetData");
        }


        public void sgetData(string s)
        {
            string[] dataflds = new string[10]; //(0 To 9);
            char[] delimters = new char[] { ' ', '\t' };
            int i,n;

            System.Console.WriteLine("sgetData ...");

            //WScript.Echo( s);

            string regResult;
            string pattern = @"^\s+|\s+$"; //trim trailing spaces
            regResult = Regex.Replace(s, pattern, "");

            dataflds.Initialize();
            string pattern2 = @"-?\d+(?:[,.]\d+)?";
            i = 0;
            foreach (Match match in Regex.Matches(regResult, pattern2))
            {
                if (match.Value != "")
                {
                    dataflds[i] = match.Value;
                    System.Console.WriteLine(match.Value);
                    i++;
                }
            }


            n = dataflds.Length;
            for (i = 0; i < n; i++)
            {
                System.Console.WriteLine(i + "<" + dataflds[i] + ">");
            }

            key = int.Parse(dataflds[0]);
            density = double.Parse(dataflds[1]);
            emod = double.Parse(dataflds[2]);
            therm = double.Parse(dataflds[3]);


            System.Console.WriteLine("... sgetData");
        }







    } //class
} //name space

File: JScript/clsPfMaterial.js
//
// Copyright (c)2014 S C Harrison
// Refer to License.txt for terms and conditions of use.
//




//Define Class
function clsPfMaterial()
{
  key =0;
  
  density = 0;        //.. density ..
  emod = 0;           //.. elastic Modulus ..
  therm = 0;          //.. coeff of thermal expansion..
}


clsPfMaterial.prototype.initialise = function()
{
  this.key = 0;
  this.density = 0;
  this.emod = 0;
  this.therm = 0;
}

clsPfMaterial.prototype.setValues = function(materialKey, massDensity, ElasticModulus, CoeffThermExpansion)
{
  this.key = materialKey
  this.density = massDensity
  this.emod = ElasticModulus
  this.therm = CoeffThermExpansion
}

clsPfMaterial.prototype.sprint = function()
{
  var s;
  
  s = ""
  s = s + StrLPad(this.key.toString(), 8);
  s = s + StrLPad(this.density.toString(), 15);
  s = s + StrLPad(this.emod.toString(), 15);
  s = s + StrLPad(this.therm.toString(), 15);

  return s;

}

clsPfMaterial.prototype.cprint = function()
{
  WScript.Echo(this.sprint());
}

clsPfMaterial.prototype.fprint = function(fp)
{
  fp.WriteLine(this.sprint());
}

clsPfMaterial.prototype.fgetData = function(fp)
{
  var s;
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "fgetData ...");
  
  s = fp.ReadLine();
  //WScript.Echo( s);
  this.sgetData(s);


   WScript.Echo( "... fgetData");
}


clsPfMaterial.prototype.sgetData = function(s)
{
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "sgetData ...");
  
   //WScript.Echo( s);

   s = s.replace(/^\s+|\s+$/gm,''); //trim trailing spaces
   dataflds = s.match(/-?\d+(?:[,.]\d+)?/gi);
   
   
   n = dataflds.length;
   for (i=0;i<n;i++)
   {
    WScript.Echo( i + "<" + dataflds[i] + ">" );
   }
   
    this.key = parseInt(dataflds[0]);
    this.density = parseFloat(dataflds[1]);
    this.emod = parseFloat(dataflds[2]);
    this.therm = parseFloat(dataflds[3]);


   WScript.Echo( "... sgetData");
}




